web/src/main/java/org/phoenixctms/ctsms/web/jersey/resource/NamedParameter.java

```diff
@@ -10,19 +12,21 @@ public NamedParameter(String name, Class<?> type) {
 		this.type = type;
 	}
 
+	@Override
+	public int hashCode() {
+		return Objects.hash(name);
+	}
+
 	@Override
 	public boolean equals(Object obj) {
-		if (obj == null) {
-			return false;
-		}
-		if (getClass() != obj.getClass()) {
+		if (this == obj)
+			return true;
+		if (obj == null)
 			return false;
-		}
-		final NamedParameter other = (NamedParameter) obj;
-		if ((this.name == null) ? (other.name != null) : !this.name.equals(other.name)) {
+		if (getClass() != obj.getClass())
 			return false;
-		}
-		return true;
+		NamedParameter other = (NamedParameter) obj;
+		return Objects.equals(name, other.name);
```

Comment:
ok for this you need changed hashCode().
hard to verify what it will break.

Reply:
agree, its hard to verify, but it's not sound to rely on a buggy implementation.